hondana/__init__.py

```diff
@@ -0,0 +1,44 @@
+import os.path
+
+import flask
+
+from .config import Configuration
+from .models import ProjectsMetadataManager
+
+
+_PROJECTS_PREFIX = "HONDANA_PROJECTS_PREFIX"
+_STORE_PREFIX = "HONDANA_STORE_PREFIX"
+
+
+def app_factory():
+    store_prefix = os.path.abspath(".store")
+
+    config = Configuration(store_prefix, "a super secret")
+    config.validate()
+
+    app = flask.Flask(__name__)
+    app.config["SECRET_KEY"] = config.secret_key
+    app.config[_PROJECTS_PREFIX] = config.projects_prefix
+    app.config[_STORE_PREFIX] = config.store_prefix
+
+    return app
+
+
+app = app_factory()
+
+def projects_prefix(app):
+    return app.config[_PROJECTS_PREFIX]
+
+
+def store_prefix(app):
+    return app.config[_STORE_PREFIX]
+
+
+@app.before_request
+def before_request():
+    flask.g.projects_metadata = ProjectsMetadataManager.from_directory(
```

Comment:
I think for now this is fine, but we may need to revisit redis later, actually. But I think adding that now is adding more complexity for not enough gain; it would be best to focus on serving the docs then optimize that IMO.

I'm thinking about a way to keep a more persistent version of this in sync on uploads, but it won't be possible across multiple requests unless it is truly global (not a thread local or context local, etc).